gammapy/data/data_store.py

```diff
@@ -335,3 +392,216 @@ def check_observations(self):
+
+    def get_events_info(self, path):
+        if path not in self._events_info:
+            self._events_info[path] = self.read_events_info(path)
+
+        return self._events_info[path]
+
+    def get_obs_info(self, path):
+        # We could add or remove info here depending on what we want in the obs table
+        return self.get_events_info(path)
+
+    @staticmethod
+    def read_events_info(filename):
+        with fits.open(filename) as hdu_list:
+            header = hdu_list["EVENTS"].header
+
+        info = {}
+        info["OBS_ID"] = header["OBS_ID"]
+        info["RA_PNT"] = header["RA_PNT"]
+        info["DEC_PNT"] = header["DEC_PNT"]
+        pos = SkyCoord(info["RA_PNT"], info["DEC_PNT"], unit="deg").galactic
+        info["GLON_PNT"] = pos.l.deg
+        info["GLAT_PNT"] = pos.b.deg
+        info["ZEN_PNT"] = 90 - float(header["ALT_PNT"])
+        info["ALT_PNT"] = header["ALT_PNT"]
+        info["AZ_PNT"] = header["AZ_PNT"]
+        info["ONTIME"] = header["ONTIME"]
+        info["LIVETIME"] = header["LIVETIME"]
+        info["DEADC"] = header["DEADC"]
+        info["TSTART"] = header["TSTART"]
+        info["TSTOP"] = header["TSTOP"]
+        info["DATE-OBS"] = header["DATE_OBS"]
+        info["TIME-OBS"] = header["TIME_OBS"]
+        info["DATE-END"] = header["DATE_END"]
+        info["TIME-END"] = header["TIME_END"]
+        info["N_TELS"] = header["N_TELS"]
+        info["OBJECT"] = header["OBJECT"]
+
+        # This is the info needed to link from EVENTS to IRFs
+        info["TELESCOP"] = header["TELESCOP"]
+        info["CALDB"] = header["CALDB"]
+        info["IRF"] = header["IRF"]
+
+        # Not part of the spec, but good to know from which file the info comes
+        info["EVENTS_FILENAME"] = filename
+        info["EVENT_COUNT"] = header["NAXIS2"]
+
+        # gti = Table.read(filename, hdu='GTI')
+        # info['GTI_START'] = gti['START'][0]
+        # info['GTI_STOP'] = gti['STOP'][0]
+
+        return info
+
+    def make_obs_table(self):
+        rows = []
+        for path in self.paths:
+            row = self.get_obs_info(path)
+            rows.append(row)
+
+        names = list(rows[0].keys())
+        table = Table(rows=rows, names=names)
+
+        table["RA_PNT"].unit = "deg"
+        table["DEC_PNT"].unit = "deg"
+        table["GLON_PNT"].unit = "deg"
+        table["GLAT_PNT"].unit = "deg"
+        table["ZEN_PNT"].unit = "deg"
+        table["ALT_PNT"].unit = "deg"
+        table["AZ_PNT"].unit = "deg"
+        table["ONTIME"].unit = "s"
+        table["LIVETIME"].unit = "s"
+        table["TSTART"].unit = "s"
+        table["TSTOP"].unit = "s"
+
+        meta = table.meta
+
+        # TODO: Values copied from one of the EVENTS headers
+        # TODO: check consistency for all EVENTS files and handle inconsistent case
+        # Transform times to first ref time? Or raise error for now?
+        # Test by combining some HESS & CTA runs?
+        meta["MJDREFI"] = 51544
```

Comment:
Hard-coding these values here seems a bit ugly. Make it possible for the user to provide these somehow? Or could they be extracted from the simulated files?

Reply:
We should definitely extract those from the EVENTS header (and check for consistency in the GTI header). We could add an option to let users set the ref time if the event file is "bad" and that info is missing. But it might be better to not put that in, forcing / encouraging people to create good EVENTS files.

If REF time is different in different EVENTS (like HESS & CTA), then it gets difficult.

I'll improve this (tomorrow or Monday) and ping you again for review of this PR.